xtask/src/set_cairo_version.rs

```diff
@@ -45,83 +46,193 @@ pub fn main(args: Args) -> Result<()> {
+    "cairo-lang-syntax",
+    "cairo-lang-syntax-codegen",
+    "cairo-lang-test-plugin",
+    "cairo-lang-test-runner",
+    "cairo-lang-test-utils",
+    "cairo-lang-utils",
+];
+
+/// Checks if the given crate is released from the starkware-libs/cairo repository.
+fn is_cairo_crate(name: &str) -> bool {
+    CAIRO_CRATES.contains(&name)
+}
 
-        // Add local path reference if requested.
-        // For local path sources, Cargo is not looking for crates recursively therefore,
-        // we need to manually provide full paths to Cairo workspace member crates.
-        if let Some(path) = &args.spec.path {
-            new_dep.insert(
-                "path",
-                path.join("crates").join(dep_name).to_string_lossy().into_owned().into(),
-            );
+/// Copies features from source dependency spec to new dependency table, if exists.
+fn copy_dependency_features(dest: &mut InlineTable, src: &Value) {
+    if let Some(dep) = src.as_inline_table() {
+        if let Some(features) = dep.get("features") {
+            dest.insert("features", features.clone());
         }
+    }
+}
 
-        // Sometimes we might specify extra features. Let's preserve these.
-        if let Some(dep) = dep.as_inline_table() {
-            if let Some(features) = dep.get("features") {
-                new_dep.insert("features", features.clone());
+/// Simplifies a `{ version = "V" }` dependency spec to shorthand `"V"` if possible.
+fn simplify_dependency_table(dep: &mut Value) {
+    *dep = match mem::replace(dep, false.into()) {
+        Value::InlineTable(mut table) => {
+            if table.len() == 1 {
+                table.remove("version").unwrap_or_else(|| table.into())
+            } else {
+                table.into()
             }
         }
 
-        // Simplify { version = "X" } to "X" if possible.
-        let new_dep: Value = if new_dep.len() == 1 {
-            new_dep.remove("version").unwrap_or_else(|| new_dep.into())
-        } else {
-            new_dep.into()
-        };
-
-        *dep = new_dep;
+        dep => dep,
     }
+}
 
-    deps.fmt();
-    deps.sort_values();
+fn purge_unused_patches(cargo_toml: &mut DocumentMut) -> Result<()> {
+    let sh = Shell::new()?;
+    let cargo_lock = sh.read_file("Cargo.lock")?.parse::<DocumentMut>()?;
 
-    eprintln!("[{table}]");
-    for (key, dep) in deps.iter().filter(|(key, _)| key.starts_with("cairo-lang-")) {
-        eprintln!("{key} = {dep}");
+    if let Some(unused_patches) = find_unused_patches(&cargo_lock) {
+        let patch =
+            cargo_toml["patch"].as_table_mut().unwrap()["crates-io"].as_table_mut().unwrap();
+
+        // Remove any patches that are not for Cairo crates.
+        patch.retain(|key, _| !unused_patches.contains(&key.to_owned()));
     }
+
+    Ok(())
+}
+
+fn find_unused_patches(cargo_lock: &DocumentMut) -> Option<Vec<String>> {
+    Some(
+        cargo_lock
+            .get("patch")?
+            .get("unused")?
```

Comment:
I don't quite understand this one, where would it come from?

Reply:
Nvm, I got it. What is the case in which it might happen though?

Reply:
Added docs answering your question